dashboard/test/models/blockly_test.rb

```diff
@@ -545,17 +545,24 @@ def create_custom_block(name, pool, block_text, args, category: 'Events')
                   </block>
                 </value>
               </block>
+              <block type="studio_ask">
+                <title name="TEXT">#{original_str}</title>
+              </block>
+              <block type="studio_showTitleScreen">
+                <title name="TITLE">#{original_str}</title>
+                <title name="TEXT">#{original_str}</title>
+              </block>
             </xml>
           </start_blocks>
         </blocks>
       </GamelabJr>
     XML
-    localized_block_xml = level.localized_text_blocks(block_xml)
+    localized_block_xml = level.localized_blocks_with_placeholder_texts(block_xml)
 
     # Expected result is an one-line XML, in which the original string
     # has been replaced by a localized string.
     block_xml_cleaned = block_xml.strip.gsub(/\s*\n\s*/, '')
-    expected_localized_block_xml = block_xml_cleaned.sub(original_str, localized_str)
+    expected_localized_block_xml = block_xml_cleaned.gsub(original_str, localized_str)
 
     assert_equal expected_localized_block_xml, localized_block_xml
```

Comment:
Does this test that there are 3 instances of `original_str`? I'm concerned it only detects that there is at least once instance of it.